enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -45,43 +66,62 @@ Organizations desire to reuse their existing enterprise log solutions to store c
+* Deploy multiple forwarder instances with separate RBAC access controls.
+* Select logs to forward based on the following criteria:
+  - built-in classification: "application", "infrastructure" "audit"
+  - k8s label selectors.
+  - by namespace.
+* Re-connect automatically if an output connection fails. 
 
 We will be successful when:
 
 * an administrator is able to deploy their own log aggregation service
-* specifies this service in log forwarding configuration
-* the service receives logs from a OKD cluster logging deployed collection agent  
+ - specifies this service in log forwarding configuration
+ - the service receives logs from a OKD cluster logging deployed collection agent  
+* a non-administrator is able to do the same within their own namespace.
 
 ### Non-Goals
 
-* Log forwarding intends to be opinionated about the source of log messages; it will explicitly define what constitutes "application logs" or "infrastructure logs". 
-It will not allow configuration of additional sources without further design considerations and how those sources may impact the collecting agent. Log forwarding is not 
-intended to provide a complex routing solution as one might achieve by using a custom collector configuration or a messaging solution (e.g. kafka).
-* It is not a goal for the tech-preview to support log forwarding outputs other then the ones identified for the goals. Admins can forward to their own fluentd via `forward` 
-and then configure that fluentd to forward to any number of specific logstore outputs
+* Will not provide complex filtering or log transformations in the GA release. The API would allow such features to be added later.
+* Will not provide complex routing other than fan-in (select multiple container log streams in a pipeline) and fan-out (send to multiple outpiuts in a pipeline) Users can forward logs to endpoints that have more complex routing features (fluentd, kafka etc.)
 * It is not a goal to provide secure storage for audit logs. If the deployer chooses to enable audit log forwarding, they need to make sure that the endpoint is compliant with governmental regulations and secure. The OpenShift logging Elasticsearch does not comply with those regulations.
+* The logging API will not expose fluentd configuration, it is intended to be independent of the underlying collector and output technologies.
 
 ## Proposal
 
-Log forwarding will provide a declarative way to specify the outputs for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more outputs. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+Log forwarding will provide a declarative 'pipeline' that associates a set of container logs with a set of outputs. The pipeline selects source logs by:
+
+* Source type: Application, Infrastructure, Audit
+* K8s label selector maps and/or expressions.
+* Namespace (for global forwarders)
```

Comment:
i'm curious how this api is going to be RBAC'd.

What distinguishes a "global" forwarder from a "namespace" forwarder?

what stops me from creating a "global" forwarder in my personal namespace?

Reply:
Spot on - the latest proposal has a cluster-scoped ClusterLogForwarder singleton for GA. Post GA there will be a separate namespace-scoped LogForwarder CRD which is restricted to forwarding  logs from it's own namespace.